### ISSUE TITLE
Binary Coss-Entropy loss

### ISSUE THREAD

**Issue by 3rd3** [dyelax/Adversarial_Video_Generation#3]
I think I've found another bug: In the [BCE loss](https://github.com/dyelax/Adversarial_Video_Generation/blob/gradient-bug/Code/loss_functions.py#L44) you calculate `-sum targets · log(preds) + (1 - targets) · log(1 - preds)`, whereas in the paper it is defined like this:

![BCE loss](https://i.imgur.com/SAtXsNQ.png)

Following the other notation in the paper, Ŷ appears to be the prediction and Y is the target, which also matches the real interval as domain for Ŷ. So it is the other way around in the paper. Is the BCE symmetric?

**Comment by 3rd3** [comment]
Also you may need to clip the values for the log as it's done [here](http://stackoverflow.com/a/33713196/852592). The base of the log should not matter, I think.

**Comment by dyelax** [comment]
Thanks for the note – this confused me at first as well, since Ŷ usually is the prediction; however, you can see in the adversarial loss equations for both D and G that the prediction is passed as the first parameter to L_bce (Y) and the 1/0 label is passed as the second (Ŷ). This lines up with the fact that, for cross-entropy, the prediction is usually inside the log. I believe they mixed up the domains, (there are several other similarly-confusing typos in the paper that the authors clarified for me when I reached out to them.)

<img width="952" alt="screen shot 2016-09-30 at 5 22 29 pm" src="https://cloud.githubusercontent.com/assets/7874539/19007445/158ea5c8-8733-11e6-8a44-c5fc5a8d4f83.png">

**Comment by 3rd3** [comment]
That also corresponds to the definition [on Wikipedia](https://en.wikipedia.org/wiki/Loss_functions_for_classification#Cross_entropy_loss). I should have looked there first. Thanks!
